lib/rototiller/task/params/command.rb

```diff
@@ -56,6 +61,37 @@ def initialize(args={}, &block)
         else
           @argument = attribute_hash[:argument]
         end
+
+      end
+
+      # adds environment variables to be tracked, messaged.
+      #   In the Command context this env_var overrides the command "name"
+      # @param [Hash] args hashes of information about the environment variable
+      # @option args [String] :name The environment variable
+      # @option args [String] :default The default value for the environment variable
+      # @option args [String] :message A message describing the use of this variable
+      #
+      # for block {|a| ... }
+      # @yield [a] Optional block syntax allows you to specify information about the environment variable, available methods match hash keys
+      def add_env(*args, &block)
+        raise ArgumentError.new("#{__method__} takes a block or a hash") if !args.empty? && block_given?
+        # this is kinda annoying we have to do this for all params? (not DRY)
+        #   have to do it this way so EnvVar doesn't become a collection
+        #   but if this gets moved to a mixin, it might be more tolerable
+        if block_given?
+          new_env_var = EnvVar.new(&block)
+          @env_vars.push(new_env_var)
+        else
+          #TODO: test this with array and non-array single hash
+          args.each do |arg| # we can accept an array of hashes, each of which defines a param
+            error_string = "#{__method__} takes an Array of Hashes. Received Array of: '#{arg.class}'"
+            raise ArgumentError.new(error_string) unless arg.is_a?(Hash)
+            new_env_var = EnvVar.new(arg)
+            @env_vars.push(new_env_var)
+          end
+        end
+        @name = @env_vars.last if @env_vars.last
+        return new_env_var
```

Comment:
yeah, we need the returns in rototiller_task because of the difference in the way we handle collections in that context.  inside of params the returns should not be required.